Nodes/HTTPSyncNode.cpp

```diff
@@ -0,0 +1,284 @@
+#include "HTTPSyncNode.h"
+#include "Messages/LocalConfigChangeMsg.h"
+#include "Messages/LocalWaypointChangeMsg.h"
+#include "Messages/ServerConfigsReceivedMsg.h"
+#include "Messages/ServerWaypointsReceivedMsg.h"
+
+/*
+
+    PURPOSE:
+        HANDLE RETRIEVAL AND WRITING TO DATABASE AUTONOMOUSLY UPON LOCAL OR SERVERSIDE CHANGES
+        PUSH LOGS ON AN INTERVAL
+        NOTIFY MESSAGEBUS WHEN NEW SERVERDATA ARRIVES
+
+*/
+
+size_t write_to_string(void *ptr, size_t size, size_t count, void *stream) {
+    ((std::string*)stream)->append((char*)ptr, 0, size*count);
+    return size*count;
+}
+
+HTTPSyncNode::HTTPSyncNode(MessageBus& msgBus, DBHandler *db, int delay, bool removeLogs)
+	:ActiveNode(NodeID::HTTPSync, msgBus), m_removeLogs(removeLogs), m_delay(delay), m_dbHandler(db)
+{
+    
+}
+
+//TODO - DESTRUCTOR?
```

Comment:
Should be ok to put the destructor in the .h file.     "virtual ~HTTPSyncNode(){}"